src/ModalComponent.php

```diff
@@ -57,7 +70,17 @@ public static function modalMaxWidth(): string
     {
         return config('livewire-ui-modal.component_defaults.modal_max_width', '2xl');
     }
+    
+    public static function modalMaxWidthClass($maxWidth = '2xl'): string
```

Comment:
Maybe change this?

```php
    public static function modalMaxWidthClass(): string
    {
        $maxWidth = self::modalMaxWidth();
        
        if(array_key_exists($maxWidth, static::$maxWidths)){
            return static::$maxWidths[$maxWidth];
        }
        
        return static::$maxWidths[static::modalMaxWidth()] ;
    }
```

Reply:
That exposes a super subtle issue, there is no real validation for this, it's not input. I was coding it in my head, but my mental model was a tiny bit off because I'm always coding for the user and not the "coder"! We are the ones calling it so we basically need to show someone where they made an illegal entry and where to fix it!  We really don't need the guard clause or defaults anymore, we are not trying to change their code, just show them what to change. 

Since we have limited them to a predefined list of keys from sm -> 7xl we should simply throw an error with a solid explanation of the issue and what to change or override to make it work

Another thing that could be done is make the array itself a config option.

I'll do that in a separate PR if you like that idea. It won't break anything.

Here is my idea a bit rough.

Leave the rest of the code as is but change this function to this as you recommended,

```php

public static function modalMaxWidthClass(): string
{
    if ( ! array_key_exists(static::modalMaxWidth(), static::$maxWidths) ) {
        
        throw new \Exception('Illegal key ' . static::modalMaxWidth() . ' not found in available protected static $maxWidths array in LivewireUI\Modal\ModalComponent. Either extend the class with your own array or choose another modalMaxWidth(). Please refer to https://github.com/wire-elements/modal/issues/26');
    }
    
    return static::$maxWidths[ static::modalMaxWidth() ];
}

```

Reply:
Thanks for the suggestion, I've made the changes in https://github.com/wire-elements/modal/pull/113 😄